allennlp/semparse/domain_languages/wikitables_language.py

```diff
@@ -462,123 +541,232 @@ def argmin(self, rows: List[Row], column: ComparableColumn) -> List[Row]:
+        if not filter_values:
+            raise ExecutionError(f"Unexpected filter value: {filter_values}")
+        if isinstance(filter_values, str):
+            filter_value = filter_values
+        elif isinstance(filter_values, list):
+            filter_value = filter_values[0]
+        else:
+            raise ExecutionError(f"Unexpected filter value: {filter_values}")
+        # Also, we need to remove the "string:" that was prepended to the entity name in the language.
         filter_value = filter_value.lstrip('string:')
-        return [row for row in rows if filter_value not in row.values[column.name]]
+        filtered_rows: List[Row] = []
+        for row in rows:
+            cell_value = row.values[column.name]
+            if isinstance(cell_value, str) and filter_value not in cell_value:
+                filtered_rows.append(row)
+        return filtered_rows
+
+    # These are some more date-column-specific functions, which only get added if we see a number
+    # column.
+    def max_date(self, rows: List[Row], column: DateColumn) -> Date:
+        """
+        Takes a list of rows and a column and returns the max of the values under that column in
+        those rows.
+        """
+        cell_values = [row.values[column.name] for row in rows]
+        if not cell_values:
+            return 0.0  # type: ignore
+        if not all([isinstance(value, Date) for value in cell_values]):
+            raise ExecutionError(f"Invalid values for date selection function: {cell_values}")
+        return max(cell_values)  # type: ignore
+
+    def min_date(self, rows: List[Row], column: DateColumn) -> Date:
+        """
+        Takes a list of rows and a column and returns the min of the values under that column in
+        those rows.
+        """
+        cell_values = [row.values[column.name] for row in rows]
+        if not cell_values:
+            return 0.0  # type: ignore
+        if not all([isinstance(value, Date) for value in cell_values]):
+            raise ExecutionError(f"Invalid values for date selection function: {cell_values}")
+        return min(cell_values)  # type: ignore
 
     # These are some more number-column-specific functions, which only get added if we see a number
     # column.
 
-    def max(self, rows: List[Row], column: NumberColumn) -> Number:
+    def max_number(self, rows: List[Row], column: NumberColumn) -> Number:
         """
         Takes a list of rows and a column and returns the max of the values under that column in
         those rows.
         """
-        cell_row_pairs = self._get_number_row_pairs_to_filter(rows, column.name)
-        if not cell_row_pairs:
+        cell_values = [row.values[column.name] for row in rows]
+        if not cell_values:
             return 0.0  # type: ignore
-        return max([value for value, _ in cell_row_pairs])  # type: ignore
+        if not all([isinstance(value, Number) for value in cell_values]):
```

Comment:
You have this error checking all over the place, except you're basically just checking that you get the types that you annotated in the function.  I don't think these help anything, and just add a bunch of noise to the code.  These checks belong as tests, not as runtime asserts (and we already have tests for things that would cause errors like this).  Unless I'm missing something, and you might actually run into this at runtime for some reason without buggy code?

Reply:
I don't remember why I had these assertions. I agree with you, and removing them doesn't seem to be breaking anything. Thanks!